src/sentry/middleware/maintenance.py

```diff
@@ -7,39 +7,16 @@
 """
 from __future__ import absolute_import, print_function
 
-import logging
-
 from django.conf import settings
 from django.http import HttpResponse
 
-logger = logging.getLogger('sentry.errors')
-
-
-DB_ERRORS = []
-
-try:
-    import MySQLdb
-except ImportError:
-    pass
-else:
-    DB_ERRORS.append(MySQLdb.OperationalError)
-
-try:
-    import psycopg2
-except ImportError:
-    pass
-else:
-    DB_ERRORS.append(psycopg2.OperationalError)
-
-DB_ERRORS = tuple(DB_ERRORS)
-
 
 class ServicesUnavailableMiddleware(object):
-    def process_request(self, request):
-        if settings.MAINTENANCE:
-            return HttpResponse('Sentry is currently in maintenance mode', status=503)
+    def __init__(self):
+        # Bail out early and disable this middleware entirely
+        if not settings.MAINTENANCE:
+            from django.core.exceptions import MiddlewareNotUsed
+            raise MiddlewareNotUsed
 
-    def process_exception(self, request, exception):
```

Comment:
I've never once seen this path hit for a database error in practice. I can revert the changes to this middleware if it's still valuable.

Even if it does work, the message is a bit misleading since it doesn't indicate maintenance mode when it's possibly just a hiccup.